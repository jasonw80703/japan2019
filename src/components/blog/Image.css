.pic {
  display: block;
  margin-left: auto;
  margin-right: auto;
}

.sm {
  max-width: 50%;
  max-height: 50%;
}

.md {
  max-width: 75%;
  max-height: 75%;
}

.lg {
  max-width: 100%;
  max-height: 100%;
}

.caption {
  color: gray;
  text-align: center;
  margin-bottom: 20px;
}

.map {
  max-width: 100%;
  max-height: 100%;
  display: block;
}

.modal-caption {
  color: gray;
}