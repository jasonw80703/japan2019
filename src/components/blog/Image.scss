@mixin img-size($size) {
  max-width: $size;
  max-height: $size;
}

.pic {
  display: block;
  margin-left: auto;
  margin-right: auto;
}

.xs {
  @include img-size(30%);
}

.sm {
  @include img-size(50%);
}

.md {
  @include img-size(75%);
}

.lg {
  @include img-size(100%);
}

.caption {
  color: gray;
  text-align: center;
  margin-bottom: 1.5%;
}

.no-caption {
  margin-bottom: 1.5%;
}