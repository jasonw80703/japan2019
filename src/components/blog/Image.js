import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ImageModal from 'components/blog/ImageModal';
import FetchingImagesSpinner from 'components/blog/FetchingImagesSpinner';
import './Image.css';

const Preload = require('react-preload').Preload;

export default class Image extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showImageModal: false,
    };

    this.handleToggleImageModal = this.handleToggleImageModal.bind(this);
  }

  handleToggleImageModal() {
    const { showImageModal } = this.state;

    this.setState({
      showImageModal: !showImageModal,
    });
  }

  render() {
    const { caption, location, source, size } = this.props;
    const { showImageModal } = this.state;
    
    const captionClass = caption ? '' : 'no-caption';
    const imageClass = `${size} pic ${captionClass}`

    return (
      <Preload
        loadingIndicator={<FetchingImagesSpinner />}
        images={[source]}
      >
        <img src={source} className={imageClass} alt='pic' onClick={this.handleToggleImageModal} />
        {caption && <p className='caption'>{caption}</p>}
        <ImageModal
          caption={caption}
          handleCloseModal={this.handleToggleImageModal}
          title={location}
          modalImage={source}
          showImageModal={showImageModal}
        />
      </Preload>
    );
  }
};

Image.defaultProps = {
  size: 'md',
};

Image.propTypes = {
  caption: PropTypes.string,
  location: PropTypes.string.isRequired,
  source: PropTypes.node.isRequired,
  size: PropTypes.string,
};
