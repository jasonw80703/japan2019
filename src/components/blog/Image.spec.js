import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Image from './Image';

Enzyme.configure({ adapter: new Adapter() });

describe('<Image />', () => {
  let wrapper;

  const defaultProps = {
    caption: 'caption caption caption',
    location: 'San Francisco',
    source: '/static/media/sf11111.jpg',
  };

  it('renders as expected', () => {
    const tree = renderer
      .create(<Image {...defaultProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('when handleToggleImageModal() is called', () => {
    wrapper = shallow(<Image {...defaultProps} />);
    wrapper.instance().handleToggleImageModal();
    expect(wrapper.state('showImageModal')).toEqual(true);
    wrapper.instance().handleToggleImageModal();
    expect(wrapper.state('showImageModal')).toEqual(false);
  });
});