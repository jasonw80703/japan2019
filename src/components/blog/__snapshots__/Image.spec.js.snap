// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> renders as expected 1`] = `
<div>
  <img
    alt="pic"
    className="md pic"
    onClick={[Function]}
    src="/static/media/sf11111.jpg"
  />
  <p
    className="caption"
  >
    caption caption caption
  </p>
</div>
`;
